baseapp/baseapp.go

```diff
@@ -890,33 +952,31 @@ func (app *BaseApp) DefaultPrepareProposal() sdk.PrepareProposalHandler {
 			byteCount int64
 		)
 
-		iterator := app.mempool.Select(ctx, req.Txs)
+		iterator := h.mempool.Select(ctx, req.Txs)
 		for iterator != nil {
 			memTx := iterator.Tx()
 
-			bz, err := app.txEncoder(memTx)
-			if err != nil {
-				panic(err)
-			}
-
+			bz, err := h.txVerifier.PrepareProposalVerifyTx(memTx)
 			txSize := int64(len(bz))
+			switch {
+			case bz == nil && errors.Is(err, ErrTxEncode):
+				panic(err)
```

Comment:
Failure to encode a transaction shouldn't result in a panic, it should just log the error, drop the bad transaction, and continue on to the next one.

Reply:
Yeah so this was discussed in the original PR. If you cannot decode or encode a tx, something very wrong happened. This should panic, however, there is a recovery mechanism for both `PrepareProposal` and `ProcessProposal`.

> Does anything prevent a malicious user from submitting invalid txs to a validator?

Technically, no.

Reply:
> If you cannot decode or encode a tx, something very wrong happened. This should panic

Why panic instead of returning an error?